zephyr/src/supp_main.c

```diff
@@ -222,6 +223,11 @@ static void iface_event_handler(struct net_mgmt_event_callback *cb,
 {
 	const char *ifname = iface->if_dev->dev->name;
 
+	if (strncmp(ifname, IFACE_MATCHING_PREFIX, strlen(IFACE_MATCHING_PREFIX)) != 0)
```

Comment:
Could use `sizeof(IFACE_MATCHING_PREFIX) - 1` instead of `strnlen` to reduce the runtime overhead